src/shared/os-util.c

```diff
@@ -31,13 +31,17 @@ int path_is_os_tree(const char *path) {
         return 1;
 }
 
-int open_os_release(const char *root, char **ret_path, int *ret_fd) {
+int open_extension_release(const char *root, const char *extension, char **ret_path, int *ret_fd) {
         _cleanup_free_ char *q = NULL;
-        const char *p;
+        char **paths, **p;
         int r, fd;
 
-        FOREACH_STRING(p, "/etc/os-release", "/usr/lib/os-release") {
-                r = chase_symlinks(p, root, CHASE_PREFIX_ROOT,
+        if (extension)
+                paths = STRV_MAKE(strjoina("/usr/lib/extension-release.d/extension-release.", extension));
```

Comment:
uh, ah, oh. mixing two kinds of stack allocation here in the same expression. Let's be careful with that. maybe this is ok on gcc, but I'd be very careful with this in the general case. 

i.e. strjoina() uses alloca() and STRV_MAKE() uses on-the-fly anonymous arrays, that is unlikely to mix well on all compilers.

given that the extension name might be long i'd probably side-step the whole thin and just use heap allocation, i.e. regular strjoin(). this is unlikely to be performance relevant after all...

Reply:
I see, I thought the issue was with function calls and not macros. TIL - will fix

Reply:
reworked a little again, as the STRV_MAKE static arrays go out of scope